assets/src/api.ts

```diff
@@ -45,6 +33,31 @@ const checkResponseStatus = (response: Response) => {
 
 const parseJson = (response: Response) => response.json()
 
+export const apiCall = <T>({
+  url,
+  parser,
+  defaultResult,
+}: {
+  url: string
+  parser: (data: any) => T
+  defaultResult?: T
+}): Promise<T> =>
+  fetch(url)
+    .then(checkResponseStatus)
+    .then(parseJson)
+    .then(({ data: data }: { data: any }) => parser(data))
+    .catch(error => {
+      if (defaultResult === undefined) {
+        // tslint:disable-next-line: no-console
+        console.error(error)
+        throw error
+      } else {
+        // tslint:disable-next-line: no-console
+        console.warn(error)
```

Comment:
We should listen to tslint and not call console here. Calling console is useless, and this is also doing what we want it to.